### PR TITLE
[2983] Fixed modern language issues whereby user cannot transverse from subject to modern language page

### PR DESCRIPTION
**Pull request by defong**
### Context
user cannot transerve from subject to modern language page

### Changes proposed in this pull request
Fixed modern language issues whereby user cannot transverse from subject to modern language page

Amended  usage of `goto_confirmation` to be in concern
Removed `course_creation_back_button` as it uses `goto_confirmation`
Amended `course_creation_back_button`to be just `govuk_back_link_to`

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by defong** [comment]
[[?] Changing a modern language course subjects from the confirmation page removes languages](https://trello.com/c/32Yo4uU1/2983-changing-a-modern-language-course-subjects-from-the-confirmation-page-removes-languages)

File: app/controllers/concerns/course_basic_detail_concern.rb [hunk 1]
```diff
@@ -145,27 +143,43 @@ def build_meta_course_creation_params
     @meta_course_creation_params = params.slice(:goto_confirmation)
   end
 
+  def continue_step
+    params[:goto_confirmation] && !%i[subjects].include?(current_step) ? :confirmation : CourseCreationStepService.new.execute(current_step: current_step, course: @course)[:next]
```
Comment (misaka) [inline app/controllers/concerns/course_basic_detail_concern.rb]: wat is this doing?

I'd break this up for the sake of legibility. For example, why `!%i[subjects].include?(current_step)` and not `current_step != :subject`, is that not the same thing?

I think it'd also benefit from being broken up onto multiple lines.

**Comment by defong** [comment]
>     * Doesn't pre-fill the modern language checkboxes from the previously selected ones

@danielburnley carded up as https://trello.com/c/8c9tBWQE/3051-split-out-modernlanguages-from-subjects

File: app/controllers/concerns/course_basic_detail_concern.rb [hunk 2]
```diff
@@ -145,27 +143,47 @@ def build_meta_course_creation_params
     @meta_course_creation_params = params.slice(:goto_confirmation)
   end
 
+  def continue_step
+    if params[:goto_confirmation] && current_step != :subjects
+      :confirmation
+    else
```
[the remark this answers is not in the document]
Comment (defong) [reply app/controllers/concerns/course_basic_detail_concern.rb]: Yeah agreed, its weird altogether...

Given that I was kinda mislead by `course_creation_back_button(@back_link_path)` 

I'm inclined to leave as is

**Pull request merged**